guja-core/src/main/java/com/wadpam/guja/oauth2/web/OAuth2Filter.java

```diff
@@ -83,28 +84,30 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
       request.setAttribute(NAME_ACCESS_TOKEN, accessToken);
 
       final DConnection conn = verifyAccessToken(accessToken);
-      if (null != conn) {
-        LOGGER.info("Authenticated");
-        LOGGER.debug(" userId = {} roles {}", conn.getUserId(), conn.getRoles());
 
-        // User is authenticated
-        request.setAttribute(NAME_CONNECTION, conn);
-        request.setAttribute(NAME_USER_ID, conn.getUserId());
-        request.setAttribute(NAME_ROLES, conn.getRoles());
+      // access_token used here for app authentication must be issued by self:
```

Comment:
Ahh, I understand.
How does the federated token gets traded for an internal token? How is the association made?
(lets have a chat later)